examples/model-viewer/main.cpp

```diff
@@ -282,16 +282,17 @@ RefPtr<Model> loadModel(
 
 struct Light : RefObject
 {
-    // A light must be able to create a shader object defining its
-    // corresponding shader type and parameter values.
-    virtual IShaderObject* createShaderObject(RendererContext* context) = 0;
+    // A light must be able to write its state into a shader parameters
+    // of the matching Slang type.
+    //
+    virtual void writeTo(ShaderCursor const& cursor) = 0;
 
     // Retrieves the shader type for this light object.
     virtual slang::TypeReflection* getShaderType(RendererContext* context) = 0;
 
     // The shader object for a light will be stashed here
     // after it is created.
-    ComPtr<IShaderObject> shaderObject;
+//    ComPtr<IShaderObject> shaderObject;
```

Comment:
I guess we can just remove this?

Reply:
Yes. I missed that one during cleanup.

One note here is that we could in theory have an application-side base class for all things that are "shader-object-able," and share that across light/material/etc. In that case the ability to create/cache a shader object could be a built-in feature of that base class (no `virtual` functions needed). E.g.:

```
// needs a real name
class ShaderObjectifiable : RefObject
{
    virtual slang::TypeReflection* getShaderType(RenderContext* context) = 0;
    virtual void writeInto(gfx::ShaderCursor const& cursor) = 0;

    IShaderObject* getShaderObject(gfx::Device* device)
    {
        if(shaderObject) return shaderObject;

        shaderObject = device->createShaderObject(getShaderType());
        writeInto(shaderObject);
        return shaderObject;
    }

protected:
    ComPtr<IShaderObject> shaderObject;
}
```